### PR TITLE
[loki-distributed] Add relabelings to ServiceMonitor

### PR DESCRIPTION
**Pull request by kamilhristov**
Example:

```yaml
serviceMonitor:
  enabled: true
  relabelings:
  - separator: /
    sourceLabels:
    - namespace
    - __meta_kubernetes_pod_label_app_kubernetes_io_component
    targetLabel: job
```

### CODE REVIEW

File: charts/loki-distributed/values.yaml [hunk 1]
```diff
@@ -181,6 +181,9 @@ serviceMonitor:
   interval: null
   # -- ServiceMonitor scrape timeout in Go duration format (e.g. 15s)
   scrapeTimeout: null
+  # -- ServiceMonitor relabel configs to apply to samples before scraping
+  # https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#relabelconfig
+  relabelings: null
```
Comment (Whyeasy) [inline charts/loki-distributed/values.yaml:186]: A list is expected, so please make this and empty list `[]`.

Comment (kamilhristov) [reply charts/loki-distributed/values.yaml]: Fixed

File: charts/loki-distributed/README.md [hunk 2]
```diff
@@ -304,6 +304,7 @@ kubectl delete statefulset RELEASE_NAME-loki-distributed-querier -n LOKI_NAMESPA
 | serviceMonitor.labels | object | `{}` | Additional ServiceMonitor labels |
 | serviceMonitor.namespace | string | `nil` | Alternative namespace for ServiceMonitor resources |
 | serviceMonitor.namespaceSelector | object | `{}` | Namespace selector for ServiceMonitor resources |
+| serviceMonitor.relabelings | string | `nil` | ServiceMonitor relabel configs to apply to samples before scraping https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#relabelconfig |
```
Comment (Whyeasy) [inline charts/loki-distributed/README.md:307]: Please rerun `helm-docs` to update with the latest change.

Comment (kamilhristov) [reply charts/loki-distributed/README.md]: Done 🤦‍♂️

**Review by Whyeasy — APPROVED** [review]
LGTM!

**Pull request merged**